lldb-eval/eval.cc

```diff
@@ -724,15 +728,25 @@ Value Interpreter::EvaluateBinaryAddition(Value lhs, Value rhs) {
 
   // Here one of the operands must be a pointer and the other one an integer.
   Value ptr, offset;
+  bool is_literal_null;
   if (lhs.IsPointer()) {
     ptr = lhs;
     offset = rhs;
+    is_literal_null = is_lhs_null;
   } else {
     ptr = rhs;
     offset = lhs;
+    is_literal_null = is_rhs_null;
   }
   assert(ptr.IsPointer() && "invalid ast: ptr must be a pointer");
   assert(offset.IsInteger() && "invalid ast: offset must be an integer");
+
+  if (is_literal_null && offset.GetInteger().isNegative()) {
+    // Binary addition with null pointer causes mismatches between LLDB and
+    // lldb-eval in some cases.
+    error_.SetUbStatus(UbStatus::kNullptrArithmetic);
+  }
```

Comment:
I think this can (and should) be checked "statically" during the parsing. Is there a reason you implemented it in the interperter?

Reply:
I can't check if the offset is negative in the parser.